src/Main.elm

```diff
@@ -0,0 +1,139 @@
+module Main exposing (..)
+
+import Html exposing (..)
+import Html.App as App
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Json.Decode as Json
+import String
+
+
+-- main
+
+main : Program Never
+main =
+  App.beginnerProgram
+    { model = emptyModel
+    , view = view
+    , update = update
+    }
+
+-- Model
+
+type alias Model =
+  { entries : List Entry
+  , input : String
+  , uid : Int
+  }
+
+type alias Entry =
+  { description : String
+  , id : Int
+  , taps : Int
+  }
+
+emptyModel : Model
+emptyModel =
+  { entries = []
+  , input = ""
+  , uid = 0
+  }
+
+
+-- Update
+
+type Msg
+  = NoOp
+  | UpdateInput String
+  | Add
+  | Tap Int
+
+
+update : Msg -> Model -> Model
+update msg model =
+  case msg of
+    NoOp -> model
+
+    UpdateInput str ->
+      { model | input = str }
+
+    Add ->
+      { model
+        | input = ""
+        , uid = model.uid + 1
+        , entries =
+            if String.isEmpty model.input then
+              model.entries
+            else
+              model.entries ++ [newEntry model.input model.uid]}
+
+    Tap id' ->
```

Comment:
It was kinda an attempt at doing that, yes … but this turns out to not be necessary. :)

I didn't think it was reserved but rather the built-in function `id` (for "identity"), which just returns whatever you pass it.

But it looks like Elm doesn't even have `id` (gotta remind myself, Elm is not Haskell), so there's no good reason to try to avoid shadowing `id`.